features/networking/operator.feature

```diff
@@ -13,3 +13,167 @@ Feature: Operator related networking scenarios
+      | resource      | clusteroperators                                                  |
+      | resource_name | network                                                           |
+      | o             | jsonpath='{.status.conditions[?(.type == "Failing")].status}'     |
+    Then the step should succeed
+    And the output should contain "False"
+    When I run the :get admin command with:
+      | resource      | clusteroperators                                                  |
+      | resource_name | network                                                           |
+      | o             | jsonpath='{.status.conditions[?(.type == "Available")].status}'   |
+    Then the step should succeed
+    And the output should contain "True"
+    """
+
+  # @author bmeng@redhat.com
+  # @case_id OCP-22202
+  @admin
+  @destructive
+  Scenario: The clusteroperator should be able to reflect the realtime status of the network when a new node added
+    Given the master version > "3.11"
+    # Check that the operator is not progressing
+    Given the expression should be true> cluster_operator('network').condition(type: 'Progressing')['status'] == "False"
+
+    # Record the original machine replica and scale it up to number +1
+    When I run the :get admin command with:
+      | resource | machineset                         |
+      | n        | openshift-machine-api              |
+      | template | {{(index .items 0).metadata.name}} |
+    Then the step should succeed
+    And evaluation of `@result[:response]` is stored in the :machineset clipboard
+    When I run the :get admin command with:
+      | resource      | machineset            |
+      | n             | openshift-machine-api |
+      | resource_name | <%= cb.machineset %>  |
+      | template      | {{.spec.replicas}}    |
+    Then the step should succeed
+    And evaluation of `@result[:response].to_i` is stored in the :original_replicas clipboard
+    And evaluation of `@result[:response].to_i + 1` is stored in the :desired_replicas clipboard
+    When I run the :scale admin command with:
+      | resource | machineset                 |
+      | name     | <%= cb.machineset %>       |
+      | replicas | <%= cb.desired_replicas %> |
+      | n        | openshift-machine-api      |
+    Then the step should succeed
+    # Scale down the machine after the scenario
+    Given I register clean-up steps:
+    """
+    When I run the :scale admin command with:
+      | resource | machineset                  |
+      | name     | <%= cb.machineset %>        |
+      | replicas | <%= cb.original_replicas %> |
+      | n        | openshift-machine-api       |
+    Then the step should succeed
+    """
+
+    # Check that the status of Progressing is truned to True during the new node provisioning
+    Given I wait up to 360 seconds for the steps to pass:
+    """
+#    And the expression should be true> `cluster_operator('network').condition(type: 'Progressing')['status']` == "True"
```

Comment:
Please fix indentation here (and possibly other similar places). We use 2 spaces for indentation. In Cucumber the keyword `Examples: ` is indented the same as the `Scenario` line. Tables are indented deeper by 2 characters. Multi-line strings are indented at the same level  as the step they are par of (not deeper).

We now have also in the project a [EditorConfig](http://editorconfig.org/) file. If provides an editor agnostic way to set some styling settings like indentation type and width and removal of trailing white spaces. Please check the linked website to find whether your editor is supported.

For generic code style rules and more information, more style information, please check [coding_guide.adoc](https://github.com/openshift/cucushift/blob/v3/doc/coding_guide.adoc)

For vim specific settings you can see [doc/vim.adoc](https://github.com/openshift/cucushift/blob/v3/doc/vim.adoc).

-- saved reply